rbd.c

```diff
@@ -239,6 +249,103 @@ static void tcmu_rbd_service_status_update(struct tcmu_device *dev,
 
 #endif /* LIBRADOS_SUPPORTS_SERVICES */
 
+#if defined LIBRBD_SUPPORTS_GETADDRS || defined RBD_LOCK_ACQUIRE_SUPPORT
+static void tcmu_rbd_rm_stale_entry_from_blacklist(struct tcmu_device *dev, struct blacklist_entry *entry)
+{
+	struct tcmu_rbd_state *state = tcmur_dev_get_private(dev);
+	const char *p = entry->addrs, *q, *end;
+	char *cmd, *addr;
+	int ret;
+
+	/*
+	 * The entry->addrs will a string like:
+	 * "[192.168.195.172:0/2203456141,192.168.195.172:0/4908756432]"
+	 * Or
+	 * "192.168.195.172:0/2203456141"
+	 */
+	while (1) {
+		if (p == NULL || *p == ']') {
+			entry->is_removed = true;
```

Comment:
Hmm -- I'm not sure it's worth keeping entries around if you fail to remove them. I was originally just suggesting to try all the cached addresses and don't just stop when you hit the first error (if any)

Reply:
Currently it will try all the cache list entries one by one without considering whether there has failure for some of them.

Yeah, it seems making no sense to try it again, so let's just remove it from the cache list even it fails.